src/app/todo/page.tsx

```diff
@@ -0,0 +1,19 @@
+import { getServerSession } from 'next-auth';
+import { redirect } from 'next/navigation';
+import { SignOutButton } from '@/components/SignIn/SignOutButton';
+
+export default async function Page() {
+  const session = await getServerSession();
+
+  if (!session) {
+    redirect('/');
+  }
+
+  return (
+    <>
+      <p>Welcome to the App, {session?.user?.name}!</p>
+      {session?.user?.image && <img src={session?.user?.image} alt="프로필" />}
+      <SignOutButton />
+    </>
```

Comment:
```suggestion
    <>
      {session?.user && (
        <>
          <p>Welcome to the App, {session.user.name}!</p>
          {session.user.image ? (
            <img src={session.user.image} alt="프로필" />
          ) : (
            <img src="~~" alt="프로필" />
          )}
        </>
      )}
      <SignOutButton />
    </>
```
아니면 유효성을 컴포넌트 최상단에 걸어주는 것도 좋을 것 같아요~